action.yml

```diff
@@ -184,16 +204,26 @@ runs:
         INPUT_POSTREMOVE: ${{ inputs.postremove }}
       run: |
         if ! fileo=$(file "${{ inputs.binary }}"); then
-          printf "could not find a binary to package"
+          printf "could not find binary: $(pwd)\n$(ls)"
           exit 1
         else
           printf "packaging binary %s" "$fileo"
         fi
-        go build -o nfpm_template .
-        INPUT_DEPENDS="${{ inputs.rpm_depends }}" ./nfpm_template > ./nfpm_rpm_config.yml
-        INPUT_DEPENDS="${{ inputs.deb_depends }}" ./nfpm_template > ./nfpm_deb_config.yml
-        cat ./nfpm_*_config.yml
-        mkdir -p ./out
-        nfpm package -f ./nfpm_rpm_config.yml -p rpm -t ./out/
-        nfpm package -f ./nfpm_deb_config.yml -p deb -t ./out/
-        ls -la ./out
+
+        package() {
+          local config_file
+          config_file="nfpm_"$1"_config.yml"
+          if ! INPUT_DEPENDS="$2" nfpm_template > "$config_file"; then
+            printf "failed to executing nfpm_template for $1"
+            exit 1
+          fi
+          cat "$config_file"
+          mkdir -p ./out
+          if ! nfpm package -f "$config_file" -p ${1} -t ./out/; then
+            printf "failed to create package with nfpm for $1 using config $(cat $config_file)\n"
+            exit 1
+          fi
```

Comment:
Nit: I don't think it's helpful to dump the config file a second time, but I do think it's helpful to always dump it.  Maybe remove the conditional and let nfpm's non-zero exit directly halt the script and job?

Reply:
I think that was an artifact of me debugging and I missed taking it out!